x/evm/keeper/keeper.go

```diff
@@ -56,8 +56,8 @@ type Keeper struct {
 	// EVM Hooks for tx post-processing
 	hooks types.EvmHooks
 
-	// custom stateless precompiled smart contracts
-	customPrecompiles evm.PrecompiledContracts
+	// Precompile extensions
+	getPrecompilesExtended func(ctx sdk.Context, evm *vm.EVM) evm.PrecompiledContracts
```

Comment:
Could make this function take an address and return a singular precompile. Opens up more modularity.